test/keycloak-connect-rest-mixed-client-restify-spec.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+
+const admin = require('./utils/realm');
+const NodeApp = require('./fixtures/node-console/restify').NodeApp;
+
+const test = require('blue-tape');
+const axios = require('axios');
+const getToken = require('./utils/token');
+
+const realmName = 'mixed-mode-realm';
+const realmManager = admin.createRealm(realmName);
+const app = new NodeApp();
+
+const auth = {
+  username: 'test-admin',
+  password: 'password'
+};
+
+test('setup', t => {
```

Comment:
Tests look good but I noticed most of the code (both test app and tests) are copy-pasted. Couldn't we somehow make it more generic? More duplicate code means higher maintenance costs.

Reply:
indeed, it was tough to get my head around, to the minimum required level to strip out the basic tests + setup and swap express with restify. I can dig deeper to think of a way to make a toggle, but I am afraid I am not familiar enough with this setup, and it will be hard to find enough time, I guess another option would be to strip it down even further? And just make sure this copy of tests will only hit that redirect at least once?

Reply:
I don't mind stripping out the restify test (and app) even more. But I'm wondering whether it wouldn't be worth it to run all the tests with restify (together with express) to prevent issues like this in the future. @abstractj Are you sure we don't support restify? Looking into the [docs](https://www.keycloak.org/docs/latest/securing_apps/index.html#_nodejs_adapter) and it says we support frameworks _like_ express.

Reply:
indeed, currently it looks like creating blockers, there was a nice suggestion (can't remember by whom) to pass in  the redirect command (at lest url) Constructed on the outer side... I just got a case failing to identify correctly http vs https ...